src/Markup/Avalonia.Markup.Xaml.Loader/CompilerExtensions/Transformers/AvaloniaXamlIlDeferredResourceTransformer.cs

```diff
@@ -51,7 +51,10 @@ private static bool ShouldBeDeferred(IXamlAstValueNode node)
             }
 
             // Do not defer resources, if it has any x:Name registration, as it cannot be delayed.
-            var nameRegistrationsVisitor = new NameScopeRegistrationVisitor(initialMetadataScopeLevel: 1);
+            // This visitor will count x:Name registrations, ignoring nested NestedScopeMetadataNode scopes.
+            // We set initial scope level to 1, assuming that this resource node is a scope of itself.
+            var nameRegistrationsVisitor = new NameScopeRegistrationVisitor(
+                initialMetadataScopeLevel: 1);
```

Comment:
I suppose we could pass zero for both?
The initial level is 0 (since it's initial by default) and we are looking for x:Name entries on the same level. The suggested change was to _replace_ `initial` parameter with `target` one for clarity.

Reply:
Yeah, true

Reply:
@kekekeks updated